torch2jax/__init__.py

```diff
@@ -270,14 +270,20 @@ def implements(torch_function, Torchishify_output=True, out_kwarg=False, Torchis
   def decorator(func):
     if out_kwarg:
 
+      def assign(a, b):
+        a.value = b
+
       def func1(*args, out=None, **kwargs):
         if out is not None:
-          out.value = func(*args, **kwargs)
+          ret = func(*args, **kwargs)
+          torch_tree_map(assign, out, ret)
```

Comment:
nit: let's add a comment here giving an example of a function that requires a non-array `out` value